AzureIotPnpDps.X/platform/cloud/cloud_service.c

```diff
@@ -221,8 +221,19 @@ static int8_t connectMQTTSocket(void)
 
 		socketState = BSD_GetSocketState(*context->tcpClientSocket);
 		if (socketState == SOCKET_CLOSED) {
-			debug_print("CLOUD: Connect socket");
-			ret = BSD_connect(*context->tcpClientSocket, (struct bsd_sockaddr*) & addr, sizeof(struct bsd_sockaddr_in));
+			debug_print("Configuring SNI to connect to %s", mqtt_host);
+
+			int optVal = 1;
+
+			ret = BSD_setsockopt(*context->tcpClientSocket, SOL_SSL_SOCKET, SO_SSL_SNI, mqtt_host, strlen(mqtt_host));
+			if (ret == BSD_SUCCESS) {
+				ret = BSD_setsockopt(*context->tcpClientSocket, SOL_SSL_SOCKET, SO_SSL_ENABLE_SNI_VALIDATION, &optVal, sizeof(optVal));
+				if (ret == BSD_SUCCESS)
+				{
+			        debug_print("CLOUD: Connect socket");
+			        ret = BSD_connect(*context->tcpClientSocket, (struct bsd_sockaddr*) & addr, sizeof (struct bsd_sockaddr_in));
+			    }
```

Comment:
nit: can't tell if this is Github but spacing here.

Reply:
Thanks! It mixed spaces with tabs. I'll convert back to tab characters.